### ISSUE TITLE
The difference between backtest and dry_run results

### ISSUE THREAD

**Issue by owlswaper** [freqtrade/freqtrade#8451]
my environment : 
* Operating system: linux ubuntu20 
* Python Version: Python 3.8.10
* CCXT version: 2.5.12
* Freqtrade Version: 2022.12

my strategy : 
In this strategy, I use leverage 5 and short trades are also done in it
Its settings too:

# ROI table:
    minimal_roi = {
        "0": 0.065,
        "15": 0.023,
        "52": 0.012,
        "109": 0
    }
    # Stoploss:
    stoploss = -0.172

    # Optimal timeframe for the strategy
    timeframe = '5m'

    can_short = True
    
    startup_candle_count = 96
    process_only_new_candles = False

    #Trailing stop:
    trailing_stop = True
    trailing_stop_positive = 0.089
    trailing_stop_positive_offset = 0.18
    trailing_only_offset_is_reached = False
    
    use_exit_signal = True
    exit_profit_only = False
    ignore_roi_if_entry_signal = False

my config : 

# config :

    "max_open_trades": 50,
    "stake_currency": "USDT",
    "stake_amount": "unlimited",
    "tradable_balance_ratio": 0.99,
    "fiat_display_currency": "USD",
    "dry_run": true,
    "cancel_open_orders_on_exit": false,
    "trading_mode": "futures",
    "margin_mode": "isolated",
    "unfilledtimeout": {
      "entry": 10,
      "exit": 10,
      "exit_timeout_count": 0,
      "unit": "minutes"

    
    "entry_pricing": {
      "price_side": "other",
      "use_order_book": true,
      "order_book_top": 1,
      "price_last_balance": 0.0,
      "check_depth_of_market": {
        "enabled": false,
        "bids_to_ask_delta": 1
      }
    },
    "exit_pricing": {
      "price_side": "other",
      "use_order_book": true,
      "order_book_top": 1
    },

    "exchange": {
      "name": "binance",
      "key": "",
      "secret": "",
      "ccxt_config": {},
      "ccxt_async_config": {},
      "pair_whitelist": [
        "BTC/USDT:USDT",
        "ETH/USDT:USDT",
        "XRP/USDT:USDT",
        "ADA/USDT:USDT",
        "MATIC/USDT:USDT",
        "FTM/USDT:USDT",
        "LINK/USDT:USDT",
        "DOT/USDT:USDT",
        "LTC/USDT:USDT",
        "ATOM/USDT:USDT"
        
      ],
      "pair_blacklist": ["BNB/.*"]
    },
    "pairlists": [{ "method": "StaticPairList" }],
  

    "api_server": {
      "enabled": true,
      "listen_ip_address": "0.0.0.0",
      "listen_port": 8080,
      "verbosity": "error",
      "enable_openapi": false,
      "jwt_secret_key": "556ebba5770ae3a07e80eda0f0f2b55df102896f8a5b86459c3433c1314345c4",
      "CORS_origins": [],
      "username": "",
      "password": ""
    },
    "bot_name": "",
    "initial_state": "running",
    "force_entry_enable": false,
    "internals": {
      "process_throttle_secs": 5
    }
  
  

my question : 

Hello, I have different results in backtest and trade live.
Of course, I am aware that the results are different between the two, but at the moment it is very much.
For example, in my backtest of the course that was done in different months: Drawdown: 1%
Win: 70%
I have received.

But in dry_run:
Drawdown: 10%
Win: 35%
I have received.

Also, I took a backtest from the previous days in which live trading was done and it was a loss, and the results were profitable.

In my opinion, the problem is with stoploss because I did not see in any output that the reason for closing trailing_stop or stoploss.

Is there a place in the backtest to see the reasons for closing positions in the backtest?

**Comment by xmatthias** [comment]
the [backtesting result](https://www.freqtrade.io/en/stable/backtesting/#understand-the-backtesting-result) will contain all information necessary to know why trades were exited.

backtesting also needs to take some [assumptions](https://www.freqtrade.io/en/stable/backtesting/#assumptions-made-by-backtesting) - which (for obvious reasons) can be exploited quite easily by a strategy, for example by looking into the future in indicators, or by setting parameters that exploit the assumptions made directly.

**Comment by owlswaper** [comment]
yes thank you
I have the results in: user_data/backtest_results
I observed that about 80% of the exits were due to trailing_stop and the rest were Stoploss. Exit_signal was almost never used in the backtest and the results were quite good.

But in the whole month of live trading, all exits were due to exit_signal, and I think Stoploss and trailing_stop do not work in this case.
How should I solve this problem?

**Comment by owlswaper** [comment]
> the [backtesting result](https://www.freqtrade.io/en/stable/backtesting/#understand-the-backtesting-result) will contain all information necessary to know why trades were exited.
> 
> backtesting also needs to take some [assumptions](https://www.freqtrade.io/en/stable/backtesting/#assumptions-made-by-backtesting) - which (for obvious reasons) can be exploited quite easily by a strategy, for example by looking into the future in indicators, or by setting parameters that exploit the assumptions made directly.

@xmatthias 
yes thank you
I have the results in: user_data/backtest_results
I observed that about 80% of the exits were due to trailing_stop and the rest were Stoploss. Exit_signal was almost never used in the backtest and the results were quite good.

But in the whole month of live trading, all exits were due to exit_signal, and I think Stoploss and trailing_stop do not work in this case.
How should I solve this problem?

**Comment by xmatthias** [comment]
Investigate what your strategy is doing differently in live vs. backtesting?

We got a list of [common mistakes](https://www.freqtrade.io/en/stable/strategy-customization/#common-mistakes-when-developing-strategies) in the docs, as well as the backtesting-assumptions listed (see link above).
